### ISSUE TITLE
What is "majority"?

### ISSUE THREAD

**Issue by nalply** [benbjohnson/thesecretlivesofdata#30]
What is "majority" exactly? More detailed questions:

1. If the network is split, will be "majority" be recalculed for each partition separately?

2. What if node A connects to nodes B and C, but node B doesn't connect to C? What is "majority" in such a case?

**Comment by DrJian** [comment]
majority means over half of all node. for example : there are 5 nodes in the cluster, majority means at least 3 nodes. when a cluster split into tow cluster A B cause network error, in terms of A or B, majority also means at least 3 nodes, for which we will get the exactly correct write data after cluster network heal.

**Comment by DrJian** [comment]
> What is "majority" exactly? More detailed questions:
> 
> 1. If the network is split, will be "majority" be recalculed for each partition separately?
> 2. What if node A connects to nodes B and C, but node B doesn't connect to C? What is "majority" in such a case?

majority means over half of all node. for example : there are 5 nodes in the cluster, majority means at least 3 nodes. when a cluster split into tow cluster A B cause network error, in terms of A or B, majority also means at least 3 nodes, for which we will get the exactly correct write data after cluster network heal.

**Comment by hubertgrzeskowiak** [comment]
Does this imply that we need an odd number of nodes, and that a partition must contain the majority of all nodes? What if there is suddenly three partitions and none has the majority?

**Comment by sid597** [comment]
> Does this imply that we need an odd number of nodes, and that a partition must contain the majority of all nodes? What if there is suddenly three partitions and none has the majority?

Yes, we need to have odd no of nodes, if we allow for even no of nodes we can run into the issue of split brain. If there are partitions such that none of them have a majority then elections will continue indefinetly and raft will not be able to respond to client.
